sdk/core/System.ClientModel/src/Internal/Logging/ClientModelLogMessages.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.NetworkInformation;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace System.ClientModel.Internal;
+
+// The methods in this class should only ever be called from LoggingHandler
+internal partial class ClientModelLogMessages
+{
+    private const int RequestEvent = 1;
+    private const int RequestContentEvent = 2;
+    private const int ResponseEvent = 5;
+    private const int ResponseContentEvent = 6;
+    private const int ResponseDelayEvent = 7;
+    private const int ErrorResponseEvent = 8;
+    private const int ErrorResponseContentEvent = 9;
+    private const int RequestRetryingEvent = 10;
+    private const int ResponseContentBlockEvent = 11;
+    private const int ErrorResponseContentBlockEvent = 12;
+    private const int ResponseContentTextEvent = 13;
+    private const int ErrorResponseContentTextEvent = 14;
+    private const int ResponseContentTextBlockEvent = 15;
+    private const int ErrorResponseContentTextBlockEvent = 16;
+    private const int RequestContentTextEvent = 17;
+    private const int ExceptionResponseEvent = 18;
+
+    private ILogger _logger;
+    public ClientModelLogMessages(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    [LoggerMessage(RequestEvent, LogLevel.Information, "Request [{requestId}] {method} {uri}\r\n{headers}client assembly: {clientAssembly}", EventName = "Request")]
```

Comment:
The source generation includes IsEnabled checks. So in this case the LogLevel will be checked twice.